adapters/httpapi_compact.c

```diff
@@ -66,9 +66,9 @@ typedef struct HTTP_HANDLE_DATA_TAG
 
 static HTTPAPI_RESULT OpenXIOConnection(HTTP_HANDLE_DATA* http_instance);
 
-/*the following function does the same as sscanf(pos2, "%d", &sec)*/
+/*the following function does the same as sscanf(pos2, "%d %s", &sec, remaining)*/
 /*this function only exists because some of platforms do not have sscanf. */
-static int ParseStringToDecimal(const char *src, int* dst)
+static int ParseStringToDecimal(const char *src, int* dst, char* remaining)
```

Comment:
After you change, this function is no longer generally parsing a decimal from string, it's specially to parse a "status code + reason" pattern from a string. You need to rename this or add a new one.

Reply:
Thanks, I renamed the function to make it clearer